### PR TITLE
[bitnami/drupal] Implement service monitor and prometheus rules

### PR DESCRIPTION
**Pull request by matthrobin**
### Description of the change

Add a service to expose metrics on specific port.
Add a service monitor to allow a prometheus operator to scrape the metrics.
Add prometheus rules template.

### Benefits

Be able to use a prometheus operator to scrape the metrics.
Be able to add custom prometheus rules without using extra deploys

### Possible drawbacks

No drawback identified

### Applicable issues

No applicable issue

### Additional information

Tested locally to prevent any regression

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by javsalgar** [comment]
Thank you so much for the PR! Could you fix the merge issues?

**Comment by matthrobin** [comment]
> Thank you so much for the PR! Could you fix the merge issues?

Done ! Thanks for the feedback

File: bitnami/drupal/templates/metrics-svc.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  {{- end }}
+  {{- if and (eq .Values.metrics.service.type "LoadBalancer") .Values.metrics.service.loadBalancerIP }}
+  loadBalancerIP: {{ .Values.metrics.service.loadBalancerIP }}
+  {{- end }}
+  {{- if and (eq .Values.metrics.service.type "LoadBalancer") .Values.metrics.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges: {{- toYaml .Values.metrics.service.loadBalancerSourceRanges | nindent 4 }}
+  {{- end }}
+  ports:
+    - name: http-metrics
+      port: {{ .Values.metrics.service.port }}
```
Comment (fmulero) [inline bitnami/drupal/templates/metrics-svc.yaml:34]:
```suggestion
      port: {{ .Values.metrics.service.ports.http-metrics }}
```

File: bitnami/drupal/values.yaml [hunk 2]
```diff
@@ -705,6 +705,91 @@ metrics:
...
     prometheus.io/port: "9117"
+  ## Drupal exporter service parameters
+  ##
+  service:
+    ## @param metrics.service.type Drupal exporter service type
+    ##
+    type: ClusterIP
+    ## @param metrics.service.port Drupal exporter service port
+    ##
+    port: 9117
```
Comment (fmulero) [inline bitnami/drupal/values.yaml:716]:
```suggestion
    ports: 
      http-metrics: 9117
```

**Review by fmulero — CHANGES_REQUESTED** [review]
Great contribution @matthrobin! thanks a lot

I added a little suggestion about the definition of the ports. We are currently setting the ports as the modifications I suggested.

File: bitnami/drupal/templates/metrics-svc.yaml (hunk 1 above)
Comment (matthrobin) [reply bitnami/drupal/templates/metrics-svc.yaml:34]: Looks good for me ! Thanks for the suggestion

File: bitnami/drupal/values.yaml (hunk 2 above)
Comment (matthrobin) [reply bitnami/drupal/values.yaml]: @fmulero Do I need to rename "http-metrics" by "metrics" in those files to be consistent with other charts ?

Comment (fmulero) [reply bitnami/drupal/values.yaml]: That could be great

Comment (matthrobin) [reply bitnami/drupal/values.yaml]: Done

**Review by fmulero — APPROVED** [review]
LGTM

**Pull request merged**